### PR TITLE
Adds fundAddress and mine functions

### PR DESCRIPTION
**Pull request by jeremy-then**

### CODE REVIEW

File: btc-transaction-helper.js
Comment (kelvinator07) [inline btc-transaction-helper.js:1]: Unit Tests?

**Review by kelvinator07 — APPROVED** [review]

**Review by nathanieliov — APPROVED** [review]

File: btc-transaction-helper.js [hunk 1]
```diff
@@ -47,31 +48,79 @@ class BtcTransactionHelper{
...
+
+    /**
+     * Transfers the btc amounts to the recipients specified in the `outputs`.
+     * Gets the spendable utxos of the `senderAddressInformation` up to the sum of the `amountInBtc`s specified in the `outputs`.
+     * @param {{address: string, privateKey: string, info?: any }} senderAddressInformation 
+     * @param {Array<{recipientAddress: string, amountInBtc: number}>} outputs 
+     * @param {Array<any>} data 
+     * @returns {string} btcTxHash
+     */
+    async transferBtc(senderAddressInformation, outputs, data) {
```
Comment (kelvinator07) [inline btc-transaction-helper.js]:
```suggestion
    async transferBtc(senderAddressInformation, recipients, data) {
```

File: btc-transaction-helper.js [hunk 2]
```diff
@@ -47,31 +48,79 @@ class BtcTransactionHelper{
...
-    async transferBtc(senderAddressInformation, receiverAddress, amountInBtc, data) {
+    /**
+     * 
+     * @param {string} address 
+     * @param {number} amountInBtc
+     * @returns 
+     */
+    async selectSpendableUTXOsFromAddress(address, amountInBtc) {
+
+        const utxos = await this.getUtxos(address);
```
Comment (kelvinator07) [inline btc-transaction-helper.js:71]:
```suggestion
        const utxos = await this.nodeClient.getUtxos(address);
```

File: btc-transaction-helper.js [hunk 3]
```diff
@@ -47,31 +48,79 @@ class BtcTransactionHelper{
...
+
+            // Adding the transfer outputs
+            outputs.forEach(output => {
+                tx.addOutput(
+                    bitcoin.address.toOutputScript(output.recipientAddress, this.btcConfig.network),
+                    this.nodeClient.btcToSatoshis(output.amountInBtc)
+                );
+            });
+
+            const actualChange = utxosInfo.change - this.btcConfig.txFee;
```
Comment (kelvinator07) [inline btc-transaction-helper.js:137]: What If The Change Is ZERO?

File: btc-transaction-helper.js [hunk 4]
```diff
@@ -47,31 +48,79 @@ class BtcTransactionHelper{
...
+
+            // Adding the change output
+            if(utxosInfo.change > 0) {
```
Comment (kelvinator07) [inline btc-transaction-helper.js]:
```suggestion
            if(utxosInfo.change > 0) {
            const actualChange = utxosInfo.change - this.btcConfig.txFee;
```

File: btc-transaction-helper.js [hunk 5]
```diff
@@ -47,31 +48,79 @@ class BtcTransactionHelper{
...
-                }
-            }
-            if (outputIndex == -1) {
-                throw new Error('funding transaction does not contain the funded adress');
+
+            const totalAmountInBtc = outputs.reduce((sum, output) => {
+                return sum + output.amountInBtc;
+            }, 0);
+
+            const fromAddress = senderAddressInformation.address;
```
Comment (kelvinator07) [inline btc-transaction-helper.js:120]: This Denotes That The `senderAddressInformation` Already Has Funds In It.. Right?

**Review by nathanieliov — APPROVED** [review]
LGTM

File: btc-transaction-helper.js (hunk 5 above)
Comment (jeremy-then) [reply btc-transaction-helper.js:120]: Yes.

File: btc-transaction-helper.js (hunk 1 above)
Comment (jeremy-then) [reply btc-transaction-helper.js]: Updated!

File: btc-transaction-helper.js (hunk 2 above)
Comment (jeremy-then) [reply btc-transaction-helper.js:71]: This `getUtxos` at the btcTxHelper level is useful to be used outside of this project, as the other functions like `mine`, etc.

File: btc-transaction-helper.js (hunk 3 above)
Comment (jeremy-then) [reply btc-transaction-helper.js:137]: Updated!

File: btc-transaction-helper.js (hunk 4 above)
Comment (jeremy-then) [reply btc-transaction-helper.js]: Updated!

**Comment by kelvinator07** [comment]
Run `npm install` so the package-lock.json file can also be updated

**Pull request merged**